molecule/provisioner/ansible/plugins/filters/molecule_core.py

```diff
@@ -36,14 +36,21 @@ def from_yaml(data):
     :return: dict
     """
     molecule_env_file = os.environ['MOLECULE_ENV_FILE']
+    molecule_base_file = os.environ['MOLECULE_BASE_FILE']
 
     env = os.environ.copy()
     env = config.set_env_from_file(env, molecule_env_file)
 
     i = interpolation.Interpolator(interpolation.TemplateWithDefaults, env)
     interpolated_data = i.interpolate(data)
 
-    loaded_data = util.safe_load(interpolated_data)
+    # When baseconfig exists merge the molecule.yml into this base
```

Comment:
Can you explain why you put this code in the `from_yaml` function? I think I have an idea but just want to check your reasoning. This function is run *a lot* when molecule is executing and it's nice to avoid touching it when possible for performance reasons.